bluemira/magnetostatics/finite_element_2d.py

```diff
@@ -0,0 +1,326 @@
+# bluemira is an integrated inter-disciplinary design tool for future fusion
+# reactors. It incorporates several modules, some of which rely on other
+# codes, to carry out a range of typical conceptual fusion reactor design
+# activities.
+#
+# Copyright (C) 2021-2023 M. Coleman, J. Cook, F. Franza, I.A. Maione, S. McIntosh,
+#                         J. Morris, D. Short
+#
+# bluemira is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 2.1 of the License, or (at your option) any later version.
+#
+# bluemira is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with bluemira; if not, see <https://www.gnu.org/licenses/>.
+
+"""
+Solver for a 2D magnetostatic problem with cylindrical symmetry
+"""
+
+from typing import Callable, Iterable, List, Optional, Tuple, Union
+
+import dolfin
+import numpy as np
+
+from bluemira.base.constants import MU_0
+
+
+def Bz_coil_axis(x: float, z: float, pz: float, current: float) -> float:
+    """
```

Comment:
@ivanmaione  please check this docstring, I dunno if I'm tired or what but I found the signature confusing

Reply:
Pretty sure you were tired... and also that the docstring and signature were confusing. Directly modified.